lapack/gonum/dtgsy2.go

```diff
@@ -0,0 +1,390 @@
+					bi.Daxpy(n-je, rhs[4], e[jsp1*lde+je+1:], 1, f[is*ldf+je+1:], 1)
+				}
+			} else if mb == 2 && nb == 1 {
+				// Build a 4x4 system Z * x = RHS.
+				z[0] = a[is*lda+is]
+				z[ldz] = a[isp1*lda+is]
+				z[2*ldz] = d[is*ldd+is]
+				z[3*ldz] = 0
+
+				z[1] = a[is*lda+isp1]
+				z[ldz+1] = a[isp1*lda+isp1]
+				z[2*ldz+1] = d[is*ldd+isp1]
+				z[3*ldz+1] = d[isp1*ldd+isp1]
+
+				z[2] = -b[js*ldb+js]
+				z[ldz+2] = 0
+				z[2*ldz+2] = -e[js*lde+js]
+				z[3*ldz+2] = 0
+
+				z[3] = 0
+				z[ldz+3] = -b[js*ldb+js]
+				z[2*ldz+3] = 0
+				z[3*ldz+3] = -e[js*lde+js]
+
+				// Set up right hand side(s).
+				rhs[0] = c[is*ldc+js]
+				rhs[1] = c[isp1*ldc+js]
+				rhs[2] = f[is*ldf+js]
+				rhs[3] = f[isp1*ldf+js]
+
+				// Solve Z * x = RHS
+				k := impl.Dgetc2(zdim, z, ldz, ipiv, jpiv)
+				if k > -1 {
+					info = k
+				}
+
+				if ijob == 0 {
+					scaloc = impl.Dgesc2(zdim, z, ldz, rhs, ipiv, jpiv)
+					if scaloc != 1 {
+						for k = 0; k < n; k++ {
+							bi.Dscal(m, scaloc, c[k:], ldc)
+							bi.Dscal(m, scaloc, f[k:], ldf)
+						}
+						scale *= scaloc
+					}
+				} else {
+					rdsum, rdscal = impl.Dlatdf(ijob, zdim, z, ldz, rhs, rdsum, rdscal, ipiv, jpiv)
+				}
+
+				// Unpack solution vectors
+				c[is*ldc+js] = rhs[0]
+				c[isp1*ldc+js] = rhs[1]
+				f[is*ldf+js] = rhs[2]
+				f[isp1*ldf+js] = rhs[3]
+
+				// Substitute R(I, J) and L(I, J) into remaining equation.
+				if i > 0 {
+					bi.Dgemv(blas.NoTrans, is, mb, -1, a[is:], lda, rhs, 1, 1, c[js:], ldc)
+					bi.Dgemv(blas.NoTrans, is, mb, -1, d[is:], ldd, rhs, 1, 1, f[js:], ldf)
+				}
+
+			} else if mb == 2 && nb == 2 {
+				// Build 8x8 system Z * x = RHS
+				impl.Dlaset('F', ldz, ldz, 0, 0, z, ldz)
+
+				z[0] = a[is*lda+is]
+				z[ldz] = a[isp1*lda+is]
+				z[4*ldz] = d[is*ldd+is]
+
+				z[1] = a[is*lda+isp1]
+				z[ldz+1] = a[isp1*lda+isp1]
+				z[4*ldz+1] = d[is*ldd+isp1]
+				z[5*ldz+1] = d[isp1*ldd+isp1]
+
+				z[2*ldz+2] = a[is*lda+is]
+				z[3*ldz+2] = a[isp1*lda+is]
+				z[6*ldz+2] = d[is*ldd+is]
+
+				z[2*ldz+3] = a[is*lda+isp1]
+				z[3*ldz+3] = a[isp1*lda+isp1]
+				z[6*ldz+3] = d[is*ldd+isp1]
+				z[7*ldz+3] = d[isp1*ldd+isp1]
+
+				z[4] = -b[js*ldb+js]
+				z[2*ldz+4] = -b[js*ldb+jsp1]
+				z[4*ldz+4] = -e[js*lde+js]
+				z[6*ldz+4] = -e[js*lde+jsp1]
+
+				z[ldz+5] = -b[js*ldb+js]
+				z[3*ldz+5] = -b[js*ldb+jsp1]
+				z[5*ldz+5] = -e[js*lde+js]
+				z[7*ldz+5] = -e[js*lde+jsp1]
+
+				z[6] = -b[jsp1*ldb+js]
+				z[2*ldz+6] = -b[jsp1*ldb+jsp1]
+				z[6*ldz+6] = -e[jsp1*lde+jsp1]
+
+				z[ldz+7] = -b[jsp1*ldb+js]
+				z[3*ldz+7] = -b[jsp1*ldb+jsp1]
+				z[7*ldz+7] = -e[jsp1*lde+jsp1]
+
+				k := 0
+				ii := mb*nb + 1
+				for jj := 0; jj < nb-1; jj++ {
+					bi.Dcopy(mb, c[is*ldc+js+jj:], ldc, rhs[k:], 1)
+					bi.Dcopy(mb, f[is*ldf+js+jj:], ldf, rhs[ii:], 1)
+					k += mb
+					ii += mb
+				}
+
+				// Solve Z * x = RHS
+				k = impl.Dgetc2(zdim, z, ldz, ipiv, jpiv)
+				if k > -1 {
```

Comment:
`> 0`?

Reply:
k is an index. Dgetc2 in LAPACK returns 0 for no index perturbed. We return -1.